cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,250 @@
+package account
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+)
+
+// Manages azd account configuration
+// `az cli` is not required and will only be called `azd` default have not already been set.
+type Manager struct {
+	azCli azcli.AzCli
+}
+
+// Creates a new Account Manager instance
+func NewManager(ctx context.Context) *Manager {
+	return &Manager{
+		azCli: azcli.GetAzCli(ctx),
+	}
+}
+
+// Gets the default subscription for the logged in account.
+// 1. Returns AZD config defaults if exists
+// 2. Returns AZ CLI defaults if exists
+func (m *Manager) GetAccountDefaults(ctx context.Context) (*config.Config, error) {
+	azdConfig := config.GetConfig(ctx)
+
+	if azdConfig.DefaultSubscription != nil && azdConfig.DefaultLocation != nil {
+		return azdConfig, nil
+	}
+
+	if azdConfig.DefaultSubscription == nil {
+		defaultSubscription, err := m.getDefaultSubscription(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving default subscription from Azure CLI: %w", err)
+		}
+
+		azdConfig.DefaultSubscription = defaultSubscription
+	}
+
+	if azdConfig.DefaultLocation == nil {
+		defaultLocation, err := m.getDefaultLocation(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving default location from Azure CLI: %w", err)
+		}
+
+		azdConfig.DefaultLocation = defaultLocation
+	}
+
+	return azdConfig, nil
+}
+
+// Gets the available Azure subscriptions for the current logged in account.
+func (m *Manager) GetSubscriptions(ctx context.Context) ([]azcli.AzCliSubscriptionInfo, error) {
+	defaultSubscription, err := m.getDefaultSubscription(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	accounts, err := m.azCli.ListAccounts(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed listing azure subscriptions: %w", err)
+	}
+
+	if defaultSubscription == nil {
+		return accounts, nil
+	}
+
+	// If default subscription is set, set it in the results
+	results := []azcli.AzCliSubscriptionInfo{}
+	for _, sub := range accounts {
+		if sub.Id == defaultSubscription.Id {
+			sub.IsDefault = true
+		}
```

Comment:
Not clear what this behavior is... If azd and az both specify IsDefault on different subscriptions, we end up with a list of two default subscriptrions?

Reply:
Agreed that if we store "defaults" that differ from "az" defaults we can get into this situation.  Azd defaults would always come first and it would  be an explicit choice by the dev to store these.  I don't recommend that we attempt to write into the az config and something like this is needed to fully decouple.  

Going to hold off on this for now.